core/src/main/java/com/datastax/oss/driver/internal/core/metrics/DropwizardMetricUpdater.java

```diff
@@ -33,46 +40,80 @@
   private static final Logger LOG = LoggerFactory.getLogger(DropwizardMetricUpdater.class);
 
   protected final Set<MetricT> enabledMetrics;
+
   protected final MetricRegistry registry;
 
-  protected DropwizardMetricUpdater(Set<MetricT> enabledMetrics, MetricRegistry registry) {
+  protected final Cache<String, Metric> metricsCache;
+
+  protected DropwizardMetricUpdater(
+      Set<MetricT> enabledMetrics, MetricRegistry registry, Ticker ticker) {
+    this(enabledMetrics, registry, ticker, Duration.ofHours(1));
+  }
+
+  protected DropwizardMetricUpdater(
+      Set<MetricT> enabledMetrics, MetricRegistry registry, Ticker ticker, Duration expiresAfter) {
     this.enabledMetrics = enabledMetrics;
     this.registry = registry;
+    metricsCache =
```

Comment:
I followed @adutra suggestion from this jira comment: https://datastax-oss.atlassian.net/browse/JAVA-2331?focusedCommentId=51936 to implement eviction of metrics using cache but I have doubts about this approach:
- How to approach configuring the eviction time? 
- This solution is eventually consistent - our users will need to tolerate some non-negligible amount of time when there will be stale metrics - are we ok with it?
- Are all metrics frequently updated? If we will pick 1 hour as a default eviction policy, can we assure that all metrics have a shorter update period?
- What if the user will set the eviction time to very short amount? (i.e. 1 second) There will be a situation in which metrics will be removed when they shouldn't. We should probably restrict the lowest value that could be set to some high value (i.e. 1 hour)
- keeping all metrics for all nodes in a cluster in the guava cache will mean that the cache can grow to thousands of entries - are we ok with that additional memory footprint?